modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/upgrade/v3_3_0/UpgradeDDMStructureLinkDLFileEntryType.java

```diff
@@ -0,0 +1,124 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.internal.upgrade.v3_3_0;
+
+import com.liferay.document.library.kernel.model.DLFileEntryMetadata;
+import com.liferay.document.library.kernel.model.DLFileEntryType;
+import com.liferay.document.library.kernel.util.DLUtil;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alicia Garcia
+ */
+public class UpgradeDDMStructureLinkDLFileEntryType extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_fixDataDefinitionIdFields();
+
+		_populateFields();
+	}
+
+	private void _fixDataDefinitionIdFields() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+			"select fileEntryTypeId, DLFileEntryType.companyId, " +
+			"dataDefinitionId from DLFileEntryType inner join DDMStructure " +
+			"ON dataDefinitionId = structureId where type_ = 0");
+
+			 PreparedStatement preparedStatement2 =
+				 AutoBatchPreparedStatementUtil.autoBatch(
+					 connection.prepareStatement(
+						 "update DLFileEntryType set dataDefinitionId = 0 " +
+						 "where fileEntryTypeId = ? "));
+
+			 PreparedStatement preparedStatement3 =
+				 AutoBatchPreparedStatementUtil.autoBatch(connection.prepareStatement(
+				 "insert into DDMStructureLink (structureLinkId, companyId, " +
+				 "classNameId, classPK, structureId) values (?, ?, ?, ?, ?)"));
+
+			ResultSet resultSet1 = preparedStatement1.executeQuery()) {
+
+			long classNameId = PortalUtil.getClassNameId(DLFileEntryType.class);
+
+			while (resultSet1.next()) {
+				long fileEntryTypeId = resultSet1.getLong(1);
+
+				preparedStatement2.setLong(1, fileEntryTypeId);
+
+				preparedStatement2.addBatch();
+
+				preparedStatement3.setLong(1, increment());
+				preparedStatement3.setLong(2, resultSet1.getLong(2));
+				preparedStatement3.setLong(3, classNameId);
+				preparedStatement3.setLong(4, fileEntryTypeId);
+				preparedStatement3.setLong(5, resultSet1.getLong(3));
+
+				preparedStatement3.addBatch();
+			}
+
+			preparedStatement2.executeBatch();
+			preparedStatement3.executeBatch();
+		}
+	}
+
+	private void _populateFields() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select uuid_, fileEntryTypeId, groupId, fileEntryTypeKey " +
```

Comment:
Why cannot we merge both methods into one? It looks wrong to me to set things to 0 in the db only to use that as select criteria. Why not use the join criteria + dataDefinitionId = 0?

Reply:
I merged it

now I have 5 PreparedStatement... seems a lot to me XD

any thougth?

Reply:
Yes, it is pretty hairy, but it is what it is.

Maybe splitting it in private methods could help, but I'm ok with it as it is.